airbyte-workers/src/test/java/io/airbyte/workers/internal/state_aggregator/StateAggregatorTest.java

```diff
@@ -0,0 +1,160 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+public class StateAggregatorTest {
+
+  StateAggregator stateAggregator;
+
+  @BeforeEach
+  public void init() {
+    stateAggregator = new DefaultStateAggregator();
+  }
+
+  @ParameterizedTest
+  @EnumSource(AirbyteStateType.class)
+  public void testCantMixType(final AirbyteStateType stateType) {
+    final List<AirbyteStateType> allTypes = Lists.newArrayList(LEGACY, GLOBAL, STREAM);
+
+    stateAggregator.ingest(getEmptyMessage(stateType));
+
+    final List<AirbyteStateType> differentTypes = allTypes.stream().filter(type -> type != stateType).toList();
+    differentTypes.forEach(differentType -> Assertions.assertThatThrownBy(() -> stateAggregator.ingest(getEmptyMessage(differentType))));
+  }
+
+  @Test
+  public void testCantMixNullType() {
+    final List<AirbyteStateType> allIncompatibleTypes = Lists.newArrayList(GLOBAL, STREAM);
+
+    stateAggregator.ingest(getEmptyMessage(null));
+
+    allIncompatibleTypes.forEach(differentType -> Assertions.assertThatThrownBy(() -> stateAggregator.ingest(getEmptyMessage(differentType))));
+
+    stateAggregator.ingest(getEmptyMessage(LEGACY));
+  }
+
+  @Test
+  public void testNullState() {
+    final AirbyteStateMessage state1 = getNullMessage(1);
+    final AirbyteStateMessage state2 = getNullMessage(2);
+
+    stateAggregator.ingest(state1);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(state1.getData()));
+
+    stateAggregator.ingest(state2);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(state2.getData()));
+  }
+
+  @Test
+  public void testLegacyState() {
+    final AirbyteStateMessage state1 = getLegacyMessage(1);
+    final AirbyteStateMessage state2 = getLegacyMessage(2);
+
+    stateAggregator.ingest(state1);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(state1.getData()));
+
+    stateAggregator.ingest(state2);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(state2.getData()));
+  }
+
+  @Test
+  public void testGlobalState() {
+    final AirbyteStateMessage state1 = getGlobalMessage(1);
+    final AirbyteStateMessage state2 = getGlobalMessage(2);
+
+    stateAggregator.ingest(state1);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(Jsons.jsonNode(List.of(state1))));
+
+    stateAggregator.ingest(state2);
+    Assertions.assertThat(stateAggregator.getAggregated()).isEqualTo(new State()
+        .withState(Jsons.jsonNode(List.of(state2))));
+  }
+
+  @Test
+  public void testStreamState() {
+    final AirbyteStateMessage state1 = getStreamMessage("a", 1);
+    final AirbyteStateMessage state2 = getStreamMessage("b", 2);
+    final AirbyteStateMessage state3 = getStreamMessage("b", 2);
```

Comment:
```suggestion
    final AirbyteStateMessage state3 = getStreamMessage("b", 3);
```
This should have a different value than state2

Reply:
Thanks I forgot to update it after the copy past